packages/salesforcedx-vscode-apex-debugger/src/index.ts

```diff
@@ -240,14 +242,38 @@ const notifyDebuggerSessionFileChanged = (): void => {
   }
 };
 
+const registerDebugHandlers = (): vscode.Disposable => {
+  const customEventHandler = vscode.debug.onDidReceiveDebugSessionCustomEvent(
+    async event => {
+      if (event?.session) {
+        const type = await getDebuggerType(event.session);
+        if (type !== DEBUGGER_TYPE) {
+          return;
+        }
+
+        if (event.event === SEND_METRIC_EVENT && isMetric(event.body)) {
```

Comment:
Seems that event.event already establishes the event as a metric event and isMetric verifies the body. Feels redundant. How can this be changed to reduce this to a single check?

Reply:
I am currently debating how to either consolidate those checks into one or eliminate the isMetric type guard. I understand that the first check for SEND_METRIC_EVENT is necessary because that is the specific event we intentionally pass from the debugger adapter to the extension for telemetry.  There might be another event type that transmits a body with a structure identical to that of SEND_METRIC_EVENT. For example, in my test, I introduced a new event type named SEND_TRIAL_EVENT, which has the same body structure as SEND_METRIC_EVENT but sends a message to telemetry of the type launchApexDebuggerTrial instead. The screenshot shows that when the first check is removed, AppInsights records both launchApexDebuggerTrial and launchApexDebugger when the ISV Debugger is initiated.  (Code changes made in the "[chore: try SEND_TRIAL_EVENT](https://github.com/forcedotcom/salesforcedx-vscode/pull/5522/commits/6aec25e9f7206ddf6d91b111be812021095fd082)" commit and then reverted.)

<img width="1728" alt="Screenshot 2024-04-05 at 10 39 55 PM" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/139700604/223165bf-bce3-48b1-bf98-a9602c736cea">
_______________________________________________

When we invoke sendEvent() in apexDebug.ts, we control the event body that we pass to the event handler. We set the event body object to contain only subject and type attributes, ensuring that the body aligns with the Metric interface's structure. The isMetric() type guard is employed to verify that there are no missing or unexpected attributes in the body sent from the debugger adapter when sendEvent() is called. However, the only objects filtered by the type guard are the ones we create in apexDebug.ts.

I am hoping to hear your suggestion on how we can best handle the checks.